src/sentry/models/outbox.py

```diff
@@ -675,6 +675,15 @@ def drain_shard(
                 if _test_processing_barrier:
                     _test_processing_barrier.wait()
 
+    @classmethod
+    def get_shard_depths_descending(cls):
+        return list(
+            {"shard_identifier": aggregate_row["shard_identifier"], "depth": aggregate_row["depth"]}
+            for aggregate_row in cls.objects.values(*cls.sharding_columns)
+            .annotate(depth=Count("pk"))
+            .order_by("-depth")
```

Comment:
Generating the dictionary and list here means that we'll fetch all of the rows from the database, which could be a few thousand rows. Could we put a limit on this and only get the first result, or have a limit parameter that lets us only fetch the results we are going to use.